gilp/simplex.py

```diff
@@ -709,19 +722,22 @@ def create_branch(lp, i, bound, branch):
 
 
 def branch_and_bound(lp: LP,
+                     int_mask: Union[np.ndarray, List, Tuple] = None,
```

Comment:
Currently, the `simplex` method assumes the `lp` is fractional and the `branch_and_bound` method assumes the `lp` is integral. Rather than specifying an `int_mask` to the `branch_and_bound` method, we should probably add this to the `LP` class itself. I don't think we can avoid this being backwards incompatible though... 

The `simplex` method can throw a custom exception if you try to call it on an LP with integral variables. We can add a `get_relaxation` method to the `LP` class that returns an LP with all integral variable replaces by fractional ones. This can be used by the branch and bound implementation.

Reply:
I agree having integrality as a property of an `LP` object would be nicer.

So, 
- move `int_mask` to `LP` 
-  do
```python
try:
    sol = simplex(lp=lp.get_relaxation(), ...)
...
frac_comp &= lp.mask
```
- change `create_branch` in BnB to pass `mask` to branch `lp`s.
- add check to `simplex`.

seems like the to-do list.

I still don't know if/what implications these may have for the visualization code because I couldn't get around to looking at it yet.

Reply:
Yeah, I think that looks right! I'll address the visualization question in the main thread.